sdk/python/foundation-models/system/inference/text-generation/llama-files/score/score.py

```diff
@@ -0,0 +1,627 @@
+import asyncio
+import json
+import logging
+import numpy as np
+import os
+
+from copy import deepcopy
+from concurrent.futures import ThreadPoolExecutor
+from inference_schema.parameter_types.abstract_parameter_type import (
+    AbstractParameterType,
+)
+from inference_schema.parameter_types.numpy_parameter_type import NumpyParameterType
+from inference_schema.parameter_types.standard_py_parameter_type import (
+    StandardPythonParameterType,
+)
+from inference_schema.schema_decorators import input_schema, output_schema
+from mlflow.models import Model
+from mlflow.pyfunc import load_model
+from mlflow.pyfunc.scoring_server import _get_jsonable_obj
+from azure.ai.mlmonitoring import Collector
+from mlflow.types.utils import _infer_schema
+from mlflow.exceptions import MlflowException
+from azure.ai.contentsafety import ContentSafetyClient
+from azure.core.credentials import AzureKeyCredential
+from azure.ai.contentsafety.models import AnalyzeTextOptions
+from aiolimiter import AsyncLimiter
+
+
+_logger = logging.getLogger(__name__)
+
+# Pandas installed, may not be necessary for tensorspec based models, so don't require it all the time
+pandas_installed = False
+try:
+    import pandas as pd
+    from inference_schema.parameter_types.pandas_parameter_type import (
+        PandasParameterType,
+    )
+
+    pandas_installed = True
+except ImportError as exception:
+    _logger.warning("Unable to import pandas")
+
+
+class AsyncRateLimitedOpsUtils:
+    # 1000 requests / 10 seconds. Limiting to 800 request per 10 secods
+    # limiting to 1000 concurrent requests
+    def __init__(
+        self,
+        ops_count=800,
+        ops_seconds=10,
+        concurrent_ops=1000,
+        thread_max_workers=1000,
+    ):
```

Comment:
How will this work if there are multiple instances as each instance will try to have it's own rate limiter. I see two options:
1. Explicitly call out in the notebook that this is for one instance only?
2. pass a paramater/env_var in the script which adjust these values based on the number of instances?

Reply:
let's chat and I will send another pr for it, trying to merge this one for experience team.